### PR TITLE
Fix dependencies and includes for FWCore and DataFormats

### PR DESCRIPTION
**Pull request by oshadura**
#### PR description:
Enable C++ modules for FWCore and FWCore (small fixes in dependencies and header includes)
cc: @vgvassilev  @davidlange6 @smuzaffar

### CODE REVIEW

File: DataFormats/HLTReco/interface/TriggerEvent.h [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #include "DataFormats/Common/interface/traits.h"
 #include <string>
 #include <vector>
+#include <assert.h>
```
Comment (Dr15Jones) [inline DataFormats/HLTReco/interface/TriggerEvent.h]:
```suggestion
#include <cassert>
```

File: DataFormats/HLTReco/interface/HLTPerformanceInfo.h [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 #include <string>
 #include <vector>
+#include <assert.h>
```
Comment (Dr15Jones) [inline DataFormats/HLTReco/interface/HLTPerformanceInfo.h]:
```suggestion
#include <cassert>
```

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

**Comment by Dr15Jones** [comment]
The pull request title doesn't seem to match what was changed.

File: DataFormats/HLTReco/interface/TriggerEvent.h (hunk 1 above)
Comment (oshadura) [reply DataFormats/HLTReco/interface/TriggerEvent.h]: Thank you! it is fixed!

File: DataFormats/HLTReco/interface/HLTPerformanceInfo.h (hunk 2 above)
Comment (oshadura) [reply DataFormats/HLTReco/interface/HLTPerformanceInfo.h]: Thank you! it is fixed!

**Comment by oshadura** [comment]
> The pull request title doesn't seem to match what was changed.
You are right, I changed it (this PR is part of https://github.com/cms-sw/cmsdist/pull/5284)

**Comment by perrotta** [comment]
-1
(wrong rebase...)

**Comment by perrotta** [comment]
please test

**Comment by davidlange6** [comment]
@oshadura and i agreed to use #28209 and #28222 instead of this PR.

**Pull request closed without merging**